### DISCUSSION TITLE
Question about converting CTF to CRC

### DISCUSSION THREAD

**Discussion by PengWei97** [mtex-toolbox/mtex#1868] [Ask Anything] [answered]
Dear MTEX Forum Experts,

I am writing to seek your guidance regarding a specific issue I encountered while attempting to convert CTF to CRC format using MTEX.

I have been using the following script:

```matlab
%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [4 4 4], 'mineral', 'Aluminum', 'color', [0.53 0.81 0.98])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'E:\SF_info\wechat\WeChat Files\wxid_ue0p3i9a3m5n22\FileStorage\File\2023-06';

% which files to be imported
fname = [pname '\1-2.ctf'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');

export_crc(ebsd,'.\1-2.crc')
```

However, I encountered the following error:

```
-> Process ebsd data before writing to files
Unrecognized field name "cprInfo".

Error in EBSD/export_crc (line 40)
cprStruct = ebsd.opt.cprInfo;

Error in a1_ctf2crc (line 32)
export_crc(ebsd,'.\1-2.crc')
```

It seems that `cprInfo` information is not added in the script. I believe this is causing the error. Could you please advise me on how to resolve this issue? Any suggestions or insights would be greatly appreciated.

Thank you in advance for your time and assistance. I look forward to hearing from you.

Best regards,
Wei

**Comment by AzdiarGazder** [comment, marked as answer]
Hi @PengWei97,

export_crc returns an error because the information located within "ebsd.opt.cprInfo" is missing.
The function cannot generate a *.cpr file because your input file is in *.ctf format. 

Please have a look at following weblink which overcomes this limitation:
https://github.com/AzdiarGazder/mtexTools/tree/main/hex2Square

This modified function should enable you to export a *.cpr and *.crc file irrespective of input file type.
Hope this helps! :)

Warm regards,
Azdi

**Reply by PengWei97** [reply]
Thank you so much. This issue has been properly resolved based on your suggestion.

**Reply by koldamendia** [reply]
Hi, @AzdiarGazder.

I am using the MtexTools exportCRC function to convert from CTF to CRC

When plotting the map using MTex, the orientation and the phases do conform, but for some reason, when I export the CRC files, the EBSD phase data seems to be distorted, corrupting the map, and rotating the orientation and Band contrast maps. 

I am attaching images of the phase and orientation maps as plotted in MTex:

![Orientations_MTex](https://github.com/mtex-toolbox/mtex/assets/58852271/7c204703-fbaf-443d-94a4-5efec6a9269d)
![Phase_MTex](https://github.com/mtex-toolbox/mtex/assets/58852271/4be65531-f01b-4fe0-9da3-d697f4ebbb60)


as plotted directly in AZtec crystal  from the .ctf file:
![Orientations_Crystal_right](https://github.com/mtex-toolbox/mtex/assets/58852271/28d895de-a629-412e-b17a-e6bc99927385)
![Phases_Crystal_right](https://github.com/mtex-toolbox/mtex/assets/58852271/121a4cb7-e9e8-42ab-85f6-f60696779d28)

and the exported .cpr file plotted in AZtec crystal.

![Orientations_Crystal_wrong](https://github.com/mtex-toolbox/mtex/assets/58852271/6f0af714-b8c8-4da1-aadf-f8f69ed54ff0)
![Phases_Crystal_wrong](https://github.com/mtex-toolbox/mtex/assets/58852271/01678213-6417-487e-a1b0-2d13eb463e05)


I have tried modifying the plotting conventions, to see if that was affecting the suggestions as to how to diagnose and correct this issue is hugely appreciated.

For context, I want to stitch the exported .cpr files in the MapStitcher  app from OI, but it doesn't recognize the .cpr exported by AZtec crystal from .ctf files, which is a different issue altogether.

Again, any help with this issue is much appreciated.

**Reply by AzdiarGazder** [reply]
Hi @koldamendia,

Thank you for the detailed post - there is indeed an issue with the function that requires resolution.
Please send me a link to download your EBSD map data.
That way, I could use your data to replicate the issue and try fixing it.

PS - Once your issue is solved, please consider using the [stitch](https://github.com/AzdiarGazder/mtexTools/tree/main/stitch) function within MTEX directly. 

Warm regards,
Azdi

**Reply by koldamendia** [reply]
Hi, @AzdiarGazder 

First of all thanks a lot for your very fast response, it is much appreciated.

I have uploaded a bunch of files corresponding to a large area map [here](https://github.com/koldamendia/EBSD_Data_example.git)

I was not aware of the stitching functionality in MTexTools - I had been looking in the MTex documentation and didn't come accross it. Thanks for pointing me in that direction. 

I have had a go, and managed to stitch two of the sections in the attached files (R1_1 and R1_2), with the idea to export them to correct for pseudosymmetry in AZtec Crystal (I can't seem to find that functionality in MTex or MTextools, but I may be missing it on account of it being called differently). However, I am still getting issues with exporting it, both to cpr (as expected, although I am getting a different error - Aztec and MapStitcher refuse to open it) but also when exporting it to ctf using export_ctf(), whereby the map is angularly distorted (except for the overlap between FoVs) - see the MTex and AZtec plots of the same data below.

![AZtec_crystal BC_stitched](https://github.com/mtex-toolbox/mtex/assets/58852271/a560660d-7a5a-4f0f-a219-cff14f7cd881)
![MTex BC_stitched](https://github.com/mtex-toolbox/mtex/assets/58852271/770445a4-9e28-4966-bd9d-93a9a13104d1)

I am aware that the latter is not related to the exportCRC function, so I assume I should be reporting it as a bug in the MTex repo as a separate Issue? Note: I have "Gridified" the stitched image before saving it to .ctf as per the note to users.

Once again, thank you very much for your assistance with this issue.

Regards,

Koldo

**Reply by AzdiarGazder** [reply]
Hi @koldamendia,

- Please note that there is a difference between the functions on the [MTEX](https://github.com/mtex-toolbox/mtex) and [mtexTools](https://github.com/AzdiarGazder/mtexTools) Github repos.
    - The first repo hosts all the functions to run MTEX natively.
    - The second repo is an independent library of extra functions that work within MTEX that I created in the course of my work. 

For example:
[export_crc](https://github.com/mtex-toolbox/mtex/blob/develop/EBSDAnalysis/%40EBSD/export_crc.m) is the MTEX's default function to export *.cpr and *.crc files.
[exportCRC](https://github.com/AzdiarGazder/mtexTools/blob/main/exportCRC/exportCRC.m) is the mtexTools version of the same function containing a few other changes that I wanted to implement for my work. 
    
In the same vein, MTEX does not have a map stitch function.
Since I needed to stitich maps together for my work, I created a [stitch](https://github.com/AzdiarGazder/mtexTools/tree/main/stitch) function and uploaded it to the mtexTools repo.

It follows that if you use any functions from the mtexTools library, the correct place to raise issues is [here](https://github.com/AzdiarGazder/mtexTools/discussions).

- Thank you so much for uploading your EBSD map data.
It was very useful to help pinpoint the issue.
I have now uploaded an edited version of the [exportCRC](https://github.com/AzdiarGazder/mtexTools/tree/main/exportCRC) function. Please download all functions in the sub-directory. 
Based on my testing of your first 2 maps, everything should work without a hitch.

- Lastly, and this is a by-the-way, if your starting off point is *.ctf files, and you wish to then undertake all analysis in OI-Channel5 (for example, to remove pseudo-symmetry and use map stitcher etc.), you can import the *.ctf file directly into the OI Program Manager. There is no need to engage MTEX or mtexTools at all.

Hope this helps.

Warm regards,
Azdi
